api/provider/aws/task_create.go

```diff
@@ -16,6 +16,22 @@ func (t *TaskProvider) Create(req models.CreateTaskRequest) (string, error) {
 	taskID := entityIDGenerator(req.TaskName)
 	fqEnvironmentID := addLayer0Prefix(t.Config.Instance(), req.EnvironmentID)
 
+	var launchType string
+	tags, err := t.TagStore.SelectByTypeAndID("environment", req.EnvironmentID)
+	if err != nil {
+		return "", err
+	}
+
+	if tag, ok := tags.WithKey("type").First(); ok {
```

Comment:
seems like this should be a helper function that both `service_create` and `task_create` can call

Reply:
Also, we should handle if the `type` tag doesn't exist and throw a meaningful error there. Currently, I assume we'd get some error from AWS and it wouldn't make any sense to the user what was going on.

Reply:
✅